packages/apps/origin-backend-irec-app/src/integration/external-device.service.ts

```diff
@@ -25,9 +25,15 @@ export class ExternalDeviceService implements IExternalDeviceService {
     public async getDeviceProductInfo({ id }: IExternalDeviceId): Promise<IProductInfo> {
         const device = await this.deviceService.findOne(id);
 
-        // TODO: proper mapping
+        const [deviceTypeCode, fuelTypeCode] = device.deviceType.split(';');
+        const deviceTypes = this.deviceService.getDeviceTypes();
+        const fuelTypes = this.deviceService.getFuelTypes();
+
+        const deviceType = deviceTypes.find((value) => value.code === deviceTypeCode)?.name ?? '';
+        const fuelType = fuelTypes.find((value) => value.code === fuelTypeCode)?.name ?? '';
+
         return {
```

Comment:
Instead of defaulting to empty string: which would eventually end up with `;` I suggest we throw the exception and make sure ask is not created.

Reply:
done